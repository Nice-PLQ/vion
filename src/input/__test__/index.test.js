import { mount } from '@vue/test-utils';
import { next } from '../../../test';
import Input from '../input.vue';

describe('Input', () => {
  // 输入框
  const wrapper = mount(Input, {
    props: {
      type: 'input',
      label: '标题',
      labelAlign: 'center',
      labelWidth: '100px',
      maxLength: 10,
      showCountIndicator: true,
      showClear: true
    }
  });

  test('input渲染', () => {
    expect(wrapper.element).toMatchSnapshot();

    const label = wrapper.find('.v-input__label');
    expect(label.text()).toBe('标题');
    expect(label.element.style.textAlign).toBe('center');
    expect(label.element.style.width).toBe('100px');
    expect(wrapper.find('.v-input__indicator')).toBeTruthy();
  });

  test('输入文本', async () => {
    const input = wrapper.find('input');
    input.element.value = 'test';
    await input.trigger('input');
    expect(wrapper.emitted('change')[0][0]).toBe('test');
  });

  test('focus && blur事件触发', async () => {
    const input = wrapper.find('input');
    input.trigger('focus');
    expect(wrapper.emitted('focus')).toBeTruthy();

    input.trigger('blur');
    expect(wrapper.emitted('blur')).toBeTruthy();
    await next(50);
    expect(wrapper.vm.isFocus).toBe(false);
  });

  test('清除文本', async () => {
    wrapper.setProps({ value: 'test' });
    await wrapper.find('input').trigger('focus');
    wrapper.find('.v-input__btn-clear').trigger('click');
    expect(wrapper.emitted('change')[1][0]).toBe('');
  });

  // 文本框
  test('textarea文本框', () => {
    const wrapper = mount(Input, {
      props: {
        type: 'textarea'
      }
    });
    expect(wrapper.element).toMatchSnapshot();
  });
});
