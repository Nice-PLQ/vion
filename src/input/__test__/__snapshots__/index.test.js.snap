// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input input渲染 1`] = `
<div
  class="v-input"
>
  <div
    class="v-input__inner"
  >
    <div
      class="v-input__label"
      style="text-align: center; width: 100px;"
    >
      
      <span>
        标题
      </span>
      
    </div>
    <div
      class="v-input__wrap"
    >
      <input
        maxlength="10"
        placeholder=""
        type="text"
      />
      <!--v-if-->
    </div>
    <!--v-if-->
  </div>
  <div
    class="v-input__help"
  >
    <!--v-if-->
    <div
      class="v-input__indicator"
    >
      10字
    </div>
  </div>
</div>
`;

exports[`Input textarea文本框 1`] = `
<div
  class="v-input v-input__textarea"
>
  <div
    class="v-input__inner"
  >
    <!--v-if-->
    <div
      class="v-input__wrap"
    >
      <textarea
        placeholder=""
        rows="3"
      />
      <!--v-if-->
    </div>
    <!--v-if-->
  </div>
  <!--v-if-->
</div>
`;
